controllers/purge_controller.go

```diff
@@ -19,180 +19,195 @@ package controllers
 import (
 	"context"
 	"fmt"
-	"strings"
 	"time"
 
+	apiextensions "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
-
-	"github.com/kyma-project/lifecycle-manager/pkg/util"
-
-	apiextensions "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	ctrlLog "sigs.k8s.io/controller-runtime/pkg/log"
 
 	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+	"github.com/kyma-project/lifecycle-manager/internal/controller/purge/metrics"
 	"github.com/kyma-project/lifecycle-manager/pkg/adapter"
 	"github.com/kyma-project/lifecycle-manager/pkg/log"
+	"github.com/kyma-project/lifecycle-manager/pkg/matcher"
 	"github.com/kyma-project/lifecycle-manager/pkg/status"
-
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
-	"k8s.io/client-go/tools/record"
-	ctrl "sigs.k8s.io/controller-runtime"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-	ctrlLog "sigs.k8s.io/controller-runtime/pkg/log"
+	"github.com/kyma-project/lifecycle-manager/pkg/util"
 )
 
 type RemoteClientResolver func(context.Context, client.ObjectKey) (client.Client, error)
 
-type CRDMatcher func(crd apiextensions.CustomResourceDefinition) bool
-
-// PurgeReconciler reconciles a Kyma object.
 type PurgeReconciler struct {
 	client.Client
 	record.EventRecorder
 	ResolveRemoteClient   RemoteClientResolver
 	PurgeFinalizerTimeout time.Duration
-	SkipCRDs              CRDMatcher
+	SkipCRDs              matcher.CRDMatcherFunc
 	IsManagedKyma         bool
 }
 
+//nolint:funlen
 func (r *PurgeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	logger := ctrlLog.FromContext(ctx)
 	logger.V(log.InfoLevel).Info("Purge Reconciliation started")
 
 	ctx = adapter.ContextWithRecorder(ctx, r.EventRecorder)
 
-	// check if kyma resource exists
 	kyma := &v1beta2.Kyma{}
 	if err := r.Get(ctx, req.NamespacedName, kyma); err != nil {
-		if !util.IsNotFound(err) {
+		if util.IsNotFound(err) {
```

Comment:
Why change this logic? The condition here is if the err is not found, then this item should enqueue again, but if it's indeed not found, then stop enqueue.

Reply:
Sure yes, reverted it. Thanks!